packages/p2p/src/components/modal-manager/modal-manager.jsx

```diff
@@ -1,9 +1,36 @@
+import React from 'react';
+import { modals } from 'Constants/modals';
 import { useModalManagerContext } from './modal-manager-context';
 
 const ModalManager = () => {
-    const { modal } = useModalManagerContext();
+    const { modal, modal_props, stacked_modal } = useModalManagerContext();
+    const { key } = modal;
 
-    return modal;
+    const Modal = modals[key];
+    const StackedModal = modals[stacked_modal?.key];
+    const getModalProps = current_modal => {
+        if (current_modal?.props && Object.keys(current_modal.props).length > 0) {
+            return current_modal.props;
+        }
+        if (modal_props.has(current_modal.key)) {
+            return modal_props.get(current_modal.key);
+        }
+        return {};
+    };
+
+    if (Modal)
+        return (
+            <React.Suspense fallback={null}>
+                <Modal {...getModalProps(modal)} />
+                {StackedModal && (
```

Comment:
`StackedModal` is needed for mobile, since in design flow for mobile, modals should be stacked, i.e. when another modal is shown, the previous modal should not be hidden but instead be behind that showed modal